etcd/src/main/scala/io/buoyant/etcd/Etcd.scala

```diff
@@ -0,0 +1,115 @@
+package io.buoyant.etcd
+
+import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
+import com.twitter.finagle.{Path, Service}
+import com.twitter.finagle.buoyant.FormParams
+import com.twitter.finagle.http.{MediaType, Message, Method, Request, Response, Status}
+import com.twitter.io.Buf
+import com.twitter.util._
+
+case class Version(etcdserver: String, etcdcluster: String)
+
+case class UnexpectedResponse(
+  method: Method,
+  uri: String,
+  params: Seq[(String, String)],
+  status: Status,
+  state: Etcd.State
+) extends Exception({
```

Comment:
I was thinking of having a parent trait or abstract class for etcd exceptions, but maybe that's not especially meaningful.